CMakeLists.txt

```diff
@@ -5,6 +5,18 @@
 
 cmake_minimum_required(VERSION 3.19.0 FATAL_ERROR)
 
+option(UNITTESTS "Run unit tests only." OFF)
+
+if(UNITTESTS)
+
+enable_testing()
+
+add_subdirectory(UNITTESTS)
+add_subdirectory(drivers/tests/UNITTESTS)
+add_subdirectory(connectivity/cellular/tests/UNITTESTS)
+
+else()
```

Comment:
without it won't it also try to configure and build mbed-os?

Reply:
yes, they need to be two different projects because we're doing cross compilation.
For our project we had it setup like this to start with and eventually moved to a completely unrelated project. We had to duplicate a few things but in the end it was much easier.

Reply:
Would it build if you do not link to Mbed Os at all?

> For our project we had it setup like this to start with and eventually moved to a completely unrelated project. We had to duplicate a few things but in the end it was much easier.

you also had either tests or the component ? Can you illustrate?